src/pages/DelegatoryValidator/StakeOperationDialog.tsx

```diff
@@ -95,6 +123,28 @@ export default function StakeOperationDialog({
     Number(balance),
   );
   const {suggestedMax, min, max} = minMax;
+  const handleClose = () => {
+    handleDialogClose();
+    setAmount("");
+  };
+
+  const [state, _] = useGlobalState();
+  const client = new AptosClient(state.network_value);
+  const [addStakeFee, setAddStakeFee] = useState<Types.MoveValue>(0);
+
+  useEffect(() => {
+    async function fetchData() {
+      if (stakeOperation === StakeOperation.STAKE) {
+        const fee = await getAddStakeFee(
+          client,
+          validator!.owner_address,
+          Number(amount).toFixed(8),
+        );
+        setAddStakeFee(fee[0]);
```

Comment:
don't have to address it necessarily in this PR but btw this data fetching logic is subject to race conditions and may display stale data.

Reply:
will address all the useQuery & race conditions in future PRs.